spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -115,6 +120,9 @@ public BigQueryTemplate(
     this.datasetName = bqDatasetName;
     this.taskScheduler = taskScheduler;
     this.bigQueryWriteClient = bigQueryWriteClient;
+    ThreadPoolTaskScheduler threadPoolTaskScheduler = (ThreadPoolTaskScheduler) taskScheduler;
```

Comment:
`taskScheduler` is "used to poll for the status of long-running BigQuery operations". I don't think it should be re-purposed for json writing as well.
Instead, I would add a setter for `jsonWriterExecutorService` and wire in a default one from `BigQueryAutoConfiguration`.

Reply:
sure, that's a better approach, I have updated the implementation accordingly, thanks